<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Obtain pre-packaged data from cBioPortal and represent as
a MultiAssayExperiment object — cBioDataPack • cBiocPortalData</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/spacelab/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Obtain pre-packaged data from cBioPortal and represent as
a MultiAssayExperiment object — cBioDataPack" />
<meta property="og:description" content="The cBioDataPack function allows the user to
download and process cancer study datasets found in MSKCC's cBioPortal.
Output datasets use the MultiAssayExperiment data
representation to faciliate analysis and data management operations." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cBiocPortalData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Quick Reference</a>
</li>
<li>
  <a href="../articles/cBioPortalData.html">User Start Guide</a>
</li>
<li>
  <a href="../articles/cBioPortalRClient.html">API Reference Guide</a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    NEWS
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/waldronlab/cBioPortalData">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Obtain pre-packaged data from cBioPortal and represent as
a MultiAssayExperiment object</h1>
    <small class="dont-index">Source: <a href='https://github.com/waldronlab/cBioPortalData/blob/master/R/cBioDataPack.R'><code>R/cBioDataPack.R</code></a></small>
    <div class="hidden name"><code>cBioDataPack.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The <code>cBioDataPack</code> function allows the user to
download and process cancer study datasets found in MSKCC's cBioPortal.
Output datasets use the MultiAssayExperiment data
representation to faciliate analysis and data management operations.</p>
    </div>

    <pre class="usage"><span class='fu'>cBioDataPack</span>(
  <span class='no'>cancer_study_id</span>,
  <span class='kw'>use_cache</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>split.field</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"Tumor_Sample_Barcode"</span>, <span class='st'>"ID"</span>),
  <span class='kw'>names.field</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"Hugo_Symbol"</span>, <span class='st'>"Entrez_Gene_Id"</span>, <span class='st'>"Gene"</span>)
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>cancer_study_id</th>
      <td><p>The cBioPortal study identifier as in
<a href='https://cbioportal.org/webAPI'>https://cbioportal.org/webAPI</a></p></td>
    </tr>
    <tr>
      <th>use_cache</th>
      <td><p>logical (default TRUE) create the default cache location
and use it to track downloaded data. If data found in the cache, data will
not be re-downloaded. A path can also be provided to data cache location.</p></td>
    </tr>
    <tr>
      <th>split.field</th>
      <td><p>A character vector of possible column names for the column
that is used to identify samples in a mutations or copy number file.</p></td>
    </tr>
    <tr>
      <th>names.field</th>
      <td><p>A character vector of possible column names for the column
that is used to label ranges from a mutations or copy number file.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A MultiAssayExperiment object</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The list of datasets can be found in the <code>studiesTable</code> dataset
by doing <code><a href='https://rdrr.io/r/utils/data.html'>data("studiesTable")</a></code>. Some datasets may not be available
for download. Please refer to the
<a href='http://cbioportal.org/data_sets.jsp'>website</a> for the full list of
datasets. Users who would like to obtain a more fine grain approach to
downloading data from the cBioPortal API, refer to the <code>cBioPortalData</code>
function.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><a href='http://cbioportal.org/data_sets.jsp'>http://cbioportal.org/data_sets.jsp</a>, <a href='cBioPortalData.html'>cBioPortalData</a></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='no'>studiesTable</span>)

<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>(<span class='no'>studiesTable</span><span class='kw'>[[</span><span class='st'>"cancer_study_id"</span>]])</div><div class='output co'>#&gt; [1] "paac_jhu_2014"       "mel_tsam_liang_2017" "all_stjude_2015"    
#&gt; [4] "all_stjude_2016"     "aml_ohsu_2018"       "laml_tcga_pub"      </div><div class='input'>
<span class='no'>mae</span> <span class='kw'>&lt;-</span> <span class='fu'>cBioDataPack</span>(<span class='st'>"acc_tcga"</span>)</div><div class='output co'>#&gt; <span class='message'>Study file in cache: acc_tcga</span></div><div class='output co'>#&gt; <span class='message'>Working on: /tmp/RtmpnA5GZ9/data_CNA.txt</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  .default = col_double(),</span>
#&gt; <span class='message'>  Hugo_Symbol = <span style='color: #BB0000;'>col_character()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>See spec(...) for full column specifications.</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  Hugo_Symbol = </span><span style='color: #BB0000;'>col_character()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>Working on: /tmp/RtmpnA5GZ9/data_RNA_Seq_v2_expression_median.txt</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  .default = col_double(),</span>
#&gt; <span class='message'>  Hugo_Symbol = </span><span style='color: #BB0000;'>col_character()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>See spec(...) for full column specifications.</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  Hugo_Symbol = </span><span style='color: #BB0000;'>col_character()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>Working on: /tmp/RtmpnA5GZ9/data_RNA_Seq_v2_mRNA_median_Zscores.txt</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  .default = col_double(),</span>
#&gt; <span class='message'>  Hugo_Symbol = </span><span style='color: #BB0000;'>col_character()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>See spec(...) for full column specifications.</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  Hugo_Symbol = </span><span style='color: #BB0000;'>col_character()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>Working on: /tmp/RtmpnA5GZ9/data_cna_hg19.seg</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  ID = </span><span style='color: #BB0000;'>col_character()</span><span>,</span>
#&gt; <span class='message'>  chrom = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  loc.start = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  loc.end = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  num.mark = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  seg.mean = </span><span style='color: #00BB00;'>col_double()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>Working on: /tmp/RtmpnA5GZ9/data_linear_CNA.txt</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  .default = col_double(),</span>
#&gt; <span class='message'>  Hugo_Symbol = </span><span style='color: #BB0000;'>col_character()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>See spec(...) for full column specifications.</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  Hugo_Symbol = </span><span style='color: #BB0000;'>col_character()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>Working on: /tmp/RtmpnA5GZ9/data_methylation_hm450.txt</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  .default = col_double(),</span>
#&gt; <span class='message'>  Hugo_Symbol = </span><span style='color: #BB0000;'>col_character()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>See spec(...) for full column specifications.</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  Hugo_Symbol = </span><span style='color: #BB0000;'>col_character()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>Working on: /tmp/RtmpnA5GZ9/data_mutations_extended.txt</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  .default = col_character(),</span>
#&gt; <span class='message'>  Entrez_Gene_Id = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  Start_Position = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  End_Position = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  t_ref_count = </span><span style='color: #BBBB00;'>col_logical()</span><span>,</span>
#&gt; <span class='message'>  t_alt_count = </span><span style='color: #BBBB00;'>col_logical()</span><span>,</span>
#&gt; <span class='message'>  n_ref_count = </span><span style='color: #BBBB00;'>col_logical()</span><span>,</span>
#&gt; <span class='message'>  n_alt_count = </span><span style='color: #BBBB00;'>col_logical()</span><span>,</span>
#&gt; <span class='message'>  Protein_position = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  Hotspot = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  TVarCov = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  Trna_ref = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  NTotCov = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  Trna_var = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  NVarCov = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  Trna_alt1 = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  Trna_alt2 = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  Trna_tot = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  TTotCov = </span><span style='color: #00BB00;'>col_double()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>See spec(...) for full column specifications.</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  .default = col_character()</span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>See spec(...) for full column specifications.</span></div><div class='output co'>#&gt; <span class='message'>Working on: /tmp/RtmpnA5GZ9/data_mutations_mskcc.txt</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  .default = col_character(),</span>
#&gt; <span class='message'>  Entrez_Gene_Id = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  Start_Position = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  End_Position = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  t_ref_count = </span><span style='color: #BBBB00;'>col_logical()</span><span>,</span>
#&gt; <span class='message'>  t_alt_count = </span><span style='color: #BBBB00;'>col_logical()</span><span>,</span>
#&gt; <span class='message'>  n_ref_count = </span><span style='color: #BBBB00;'>col_logical()</span><span>,</span>
#&gt; <span class='message'>  n_alt_count = </span><span style='color: #BBBB00;'>col_logical()</span><span>,</span>
#&gt; <span class='message'>  Protein_position = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  Hotspot = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  TVarCov = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  Trna_ref = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  NTotCov = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  Trna_var = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  NVarCov = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  Trna_alt1 = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  Trna_alt2 = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  Trna_tot = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  TTotCov = </span><span style='color: #00BB00;'>col_double()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>See spec(...) for full column specifications.</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  .default = col_character()</span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>See spec(...) for full column specifications.</span></div><div class='output co'>#&gt; <span class='message'>Working on: /tmp/RtmpnA5GZ9/data_mutsig.txt</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  .default = col_double(),</span>
#&gt; <span class='message'>  gene = </span><span style='color: #BB0000;'>col_character()</span><span>,</span>
#&gt; <span class='message'>  description = </span><span style='color: #BB0000;'>col_character()</span><span>,</span>
#&gt; <span class='message'>  p = </span><span style='color: #BB0000;'>col_character()</span><span>,</span>
#&gt; <span class='message'>  q = </span><span style='color: #BB0000;'>col_character()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>See spec(...) for full column specifications.</span></div><div class='output co'>#&gt; <span class='message'>Working on: /tmp/RtmpnA5GZ9/data_rppa.txt</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  .default = col_double(),</span>
#&gt; <span class='message'>  Composite.Element.REF = </span><span style='color: #BB0000;'>col_character()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>See spec(...) for full column specifications.</span></div><div class='output co'>#&gt; <span class='message'>Working on: /tmp/RtmpnA5GZ9/data_rppa_Zscores.txt</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  .default = col_double(),</span>
#&gt; <span class='message'>  Composite.Element.REF = </span><span style='color: #BB0000;'>col_character()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>See spec(...) for full column specifications.</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  .default = col_character(),</span>
#&gt; <span class='message'>  INITIAL_PATHOLOGIC_DX_YEAR = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  WEISS_SCORE_OVERALL = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  MITOSES_PER_50_HPF = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  AGE = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  DAYS_TO_INITIAL_PATHOLOGIC_DIAGNOSIS = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  OS_MONTHS = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  DFS_MONTHS = </span><span style='color: #00BB00;'>col_double()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>See spec(...) for full column specifications.</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  .default = col_character(),</span>
#&gt; <span class='message'>  DAYS_TO_COLLECTION = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  SAMPLE_INITIAL_WEIGHT = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  OCT_EMBEDDED = </span><span style='color: #BBBB00;'>col_logical()</span><span>,</span>
#&gt; <span class='message'>  SAMPLE_TYPE_ID = </span><span style='color: #00BB00;'>col_double()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>See spec(...) for full column specifications.</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  .default = col_character(),</span>
#&gt; <span class='message'>  DAYS_TO_COLLECTION = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  SAMPLE_INITIAL_WEIGHT = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  OCT_EMBEDDED = </span><span style='color: #BBBB00;'>col_logical()</span><span>,</span>
#&gt; <span class='message'>  SAMPLE_TYPE_ID = </span><span style='color: #00BB00;'>col_double()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>See spec(...) for full column specifications.</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  .default = col_character()</span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='message'>See spec(...) for full column specifications.</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  index = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  chromosome = </span><span style='color: #BB0000;'>col_character()</span><span>,</span>
#&gt; <span class='message'>  region_start = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  region_end = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  peak_start = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  peak_end = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  enlarged_peak_start = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  enlarged_peak_end = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  n_genes_in_region = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  genes_in_region = </span><span style='color: #BB0000;'>col_character()</span><span>,</span>
#&gt; <span class='message'>  n_genes_in_peak = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  genes_in_peak = </span><span style='color: #BB0000;'>col_character()</span><span>,</span>
#&gt; <span class='message'>  n_genes_on_chip = </span><span style='color: #BB0000;'>col_character()</span><span>,</span>
#&gt; <span class='message'>  genes_on_chip = </span><span style='color: #BB0000;'>col_character()</span><span>,</span>
#&gt; <span class='message'>  `top 3` = </span><span style='color: #BB0000;'>col_character()</span><span>,</span>
#&gt; <span class='message'>  amp = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  cytoband = </span><span style='color: #BB0000;'>col_character()</span><span>,</span>
#&gt; <span class='message'>  q_value = </span><span style='color: #00BB00;'>col_double()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='warning'>Warning:  Multiple prefixes found, using keyword 'region' or taking first one</span></div><div class='output co'>#&gt; <span class='message'>Parsed with column specification:</span>
#&gt; <span class='message'>cols(</span>
#&gt; <span class='message'>  index = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  chromosome = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  region_start = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  region_end = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  peak_start = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  peak_end = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  enlarged_peak_start = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  enlarged_peak_end = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  n_genes_in_region = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  genes_in_region = </span><span style='color: #BB0000;'>col_character()</span><span>,</span>
#&gt; <span class='message'>  n_genes_in_peak = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  genes_in_peak = </span><span style='color: #BB0000;'>col_character()</span><span>,</span>
#&gt; <span class='message'>  n_genes_on_chip = </span><span style='color: #BB0000;'>col_character()</span><span>,</span>
#&gt; <span class='message'>  genes_on_chip = </span><span style='color: #BB0000;'>col_character()</span><span>,</span>
#&gt; <span class='message'>  `top 3` = </span><span style='color: #BB0000;'>col_character()</span><span>,</span>
#&gt; <span class='message'>  amp = </span><span style='color: #00BB00;'>col_double()</span><span>,</span>
#&gt; <span class='message'>  cytoband = </span><span style='color: #BB0000;'>col_character()</span><span>,</span>
#&gt; <span class='message'>  q_value = </span><span style='color: #00BB00;'>col_double()</span><span></span>
#&gt; <span class='message'>)</span></div><div class='output co'>#&gt; <span class='warning'>Warning:  Multiple prefixes found, using keyword 'region' or taking first one</span></div><div class='input'>
</div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Levi Waldron, Marcel Ramos.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


